src/main/resources/migrations/00043_create_table_oauth_states.sql

```diff
@@ -0,0 +1,15 @@
+--liquibase formatted sql
+
+--changeset uk.gov.pay:add_table-oauth_states
+CREATE TABLE oauth_states (
```

Comment:
Can we come up with a more descriptive name for this?

I understand these are effectively tokens or identifiers we use during the partner connect flow so something that tries to capture that?

OAuth is just the mechanism used for connecting but doesn’t really matter from the database perspective and “state” can mean many things (I assume its name comes from the name of the parameter GoCardless use).

Reply:
I used `oauth_tokens`. I know it doesn't matter for the db but I think it matters to the developer who is going to look at the name of the table and try to figure out what is doing.
I believe `oauth_tokens` is misleading because this might confuse people in thinking that this token is the one we get from GoCardless, which is not true. 
This table unfortunately is again very specific to GoCardless so we might as well call it `gocardless_oauth_request_tokens`.

Any naming suggestions are more than welcome :)

Reply:
It’s already very specific to GoCardless, so a GoCardless-specific name is fine. Call it `gocardless_partner_app_account_connect_tokens` something (`tokens` is unfortunately a very overloaded term: GoCardless’s own docs recommend the `state` parameter should have a “CSRF token” in it).

This table contains stuff related to connecting GoCardless accounts to our partner app (GoCardless unfortunately call this process “Connecting your users’ accounts” — “users” is such an overloaded term!) so I think it’s a good idea to try to capture in the name that it’s related to that process (`gocardless_partner_app_account_connect` is the best I can come up with right now).

Not keen on having `oauth` in the name: OAuth is just the mechanism used for the partner app account connect stuff. When we connect accounts, we’re not doing OAuth for the sake of OAuth, we’re doing it in order to connect accounts to our partner app. It’s conceivable we may end up with other OAuth functionality in our app in the future to do other things, which might mean a table name like `oauth_tokens` could cause confusion (also OAuth already has a concept of tokens, which could lead to more confusion).